src/gui/folder.cpp

```diff
@@ -783,6 +824,21 @@ void Folder::slotSyncFinished(bool success)
         journalDb()->setSelectiveSyncList(SyncJournalDb::SelectiveSyncWhiteList, QStringList());
     }
 
+    // bug: This function uses many different criteria for "sync was successful" - investigate!
+    if ((_syncResult.status() == SyncResult::Success
+            || _syncResult.status() == SyncResult::Problem)
+        && success) {
+        if (_engine->lastLocalDiscoveryStyle() == LocalDiscoveryStyle::FilesystemOnly) {
+            _timeSinceLastFullLocalDiscovery.start();
+        } else if (_syncResult.status() == SyncResult::Success) {
+            // Optimization opportunity: Currently we wipe this only
+            // on 100% successful sync. ('Problem' syncs could have the touched
+            // file trigger a soft-error-and-retry scenario and wiping this list
+            // would mean the file isn't retried)
+            _locallyModifiedDirs.clear();
```

Comment:
What if there were change hapenning during the sync!

Maybe we should clear it as we start the sync, and if the sync fails for any problem, we add them back.

Reply:
@ogoffart Yes, great catch! That means we really need to remove files from this list as the individual jobs report success even in the initial version. At least that also helps with the issue regarding SoftErrors.